### ISSUE TITLE
Failure of testing due to different levels of optimization when building a project.

### ISSUE THREAD

**Issue by vel21ripn** [ntop/nDPI#465]
Building a project with the CFLAGS="-g -O0" and CFLAGS="-g -O2" keys produces different results.
GCC-5.3.0 (64bit) is used.
PR #464  is the first step in identifying the problem.

**Comment by kYroL01** [comment]
Actually I don't understand the correlation between the pull request (that I think is correct) and the different levels of optimizzation. Can you explain better ? Thank you

**Comment by vel21ripn** [comment]
Previously, sorting used only the total number of bytes in the stream, which gave, in the presence of undefined program behavior, a different order of output of the list of flows.
PR #464 eliminates this uncertainty in the behavior of the program.
Different results of the program with different optimization is a special case for different results when working on different architectures.
```
make distclean
./autogen.sh
CFLAGS="-g O2" ./configure
make -j4
cd tests && ./do.sh
```
OK
```
make distclean
./autogen.sh
CFLAGS="-g O0" ./configure
make -j4
( cd tests && ./do.sh ) 
```
Not OK

All differences is Skype detection.
```
skype_no_unknown.pcap                   ERROR
../example/ndpiReader -p ../example/protos.txt -q -i pcap/skype_no_unknown.pcap -w /tmp/reader.out -v 1
1c1
< Unknown       163     59526   10
---
> Unknown       186     61791   12
10c10
< Skype 1314    192401  225
---
> Skype 1291    190136  223
24d23
<       TCP 192.168.1.34:51314 <-> 93.79.224.176:14506 [proto: 125/Skype][11 pkts/1407 bytes <-> 9 pkts/652 bytes]
116d114
<       TCP 192.168.1.34:51300 <-> 76.167.161.6:20274 [proto: 125/Skype][2 pkts/132 bytes <-> 1 pkts/74 bytes]
278a277
>       TCP 192.168.1.34:51314 <-> 93.79.224.176:14506 [proto: 0/Unknown][11 pkts/1407 bytes <-> 9 pkts/652 bytes]
282a282
>       TCP 192.168.1.34:51300 <-> 76.167.161.6:20274 [proto: 0/Unknown][2 pkts/132 bytes <-> 1 pkts/74 bytes]
skype.pcap                              ERROR
../example/ndpiReader -p ../example/protos.txt -q -i pcap/skype.pcap -w /tmp/reader.out -v 1
8c8
< SSL   88      8268    6
---
> SSL   96      8876    7
10c10
< Skype 2147    325017  250
---
> Skype 2139    324409  249
125c125
<       TCP 192.168.1.34:50146 -> 157.56.53.51:443 [proto: 91.125/SSL.Skype][8 pkts/608 bytes -> 0 pkts/0 bytes]
---
>       TCP 192.168.1.34:50146 -> 157.56.53.51:443 [proto: 91/SSL][8 pkts/608 bytes -> 0 pkts/0 bytes]
```
There is another question: what prevents to take samples of traffic from https://svn.ntop.org/svn/ntop/trunk/nDPIng/example/flows/? I can issue PR.
Samples should be checked for correctness. There definitely are some bad examples.

**Comment by kYroL01** [comment]
First of all, I checked and IMHO the corrupted samples (_nintendo_ and _skype-conference-call_) must be deleted.

**Comment by vel21ripn** [comment]
Error found and fixed ( #466 )
